rave-compiler/src/main/java/com/uber/rave/compiler/RaveWriter.java

```diff
@@ -156,7 +157,7 @@ private MethodSpec generateConstructor(@NonNull RaveIR raveIR) {
         builder.endControlFlow();
         List<MethodSpec> concreteSpecs = new ArrayList<>(raveIR.getNumClasses());
         for (ClassIR classIR : raveIR.getClassIRs()) {
-            MethodSpec specificSpec = generateConcreteMethodSpec(classIR);
+            MethodSpec specificSpec = generateConcreteMethodSpec(raveIR, classIR);
```

Comment:
We could scope this down to only the mode. However I thought using `RaveIR` was more clear because it makes it explicit that things at the library level are used to generate the method that performs validation. When more things are added on a per library level in the future (like supporting different kinds of nullness annotations) that information should live in RaveIR and not end up as a long list of parameters.